### PR TITLE
[REFACTOR] Simplify builder chaining code codegen

### PR DESCRIPTION
**Pull request by pcrosemurgy**
* **Tickets addressed:** AERIE-2005
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
"Hotfix" follow-up to simplify some of the `ActivityTypes.Builder` code generation. Motivated by https://github.com/NASA-AMMOS/aerie/pull/281#discussion_r943981202. This PR does not change the generated code, it just organizes the existing code generation logic to be easier to follow and more logical.

### CODE REVIEW

File: merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java [hunk 1]
```diff
@@ -612,24 +612,18 @@ private static TypeSpec generateActivityTypesBuilder(final MissionModelRecord mi
   private static MethodSpec generateActivityTypesRegisterMethod(final MissionModelRecord missionModel) {
     final var typeName = missionModel.getTypesName();
 
-    final var cb = CodeBlock.builder()
-        .add(
-            "return new $T(new $T()",
-            typeName,
-            typeName.nestedClass("Builder"));
-
+    final var cb = CodeBlock.builder();
```
Comment (Twisol) [inline merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java]: I'd love a better name than `cb` 😺

**Review by Twisol — APPROVED** [review]
Great! One minor comment, but this makes it much easier for me to see how the generated code is composed.

File: merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java (hunk 1 above)
Comment (pcrosemurgy) [reply merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java]: Haha yeah this is tricky. It's an `ActivityTypes.Builder` code block builder 🥴 Open to suggestions.

Comment (pcrosemurgy) [reply merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java]: Maybe just `builderChain`

Comment (Twisol) [reply merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java]: Yeah, `activityTypesBuilderChain` if we feel like being verbose. (Honestly, in this kind of templating setup, I think being verbose with names is a very good thing.)

Comment (pcrosemurgy) [reply merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java]: Nice, just pushed!

**Comment by pcrosemurgy** [comment]
Not sure why we're hitting this failure:
```
Action was triggered by pull_request: using SHA from head of source branch
Check runs will be created with SHA=90bcc1[18](https://github.com/NASA-AMMOS/aerie/runs/7796875758?check_suite_focus=true#step:14:19)3eee4dca067aa2b32ec69cc118e954a7
Listing all files tracked by git
Found 950 files tracked by GitHub
Using test report parser 'java-junit'
Creating test report ☕ JUnit Results
Error: No test report files were found
```

**Comment by skovati** [comment]
Screen Shot 2022-08-11 at 5 35 39 PM
Looks like services aren't even starting because it can't find... `curl`?

**Comment by pcrosemurgy** [comment]
@skovati Good find! Not sure what Dockerfile is in charge of this but maybe there's a missing `apt-get update`?

**Comment by skovati** [comment]
That's definitely my guess, strange that this just shows up now

Looks like it's in the merlin-server Dockerfile

**Comment by pcrosemurgy** [comment]
Thanks @skovati, just pushed a change to that Dockerfile.

**Comment by skovati** [comment]
Looks like this is affecting others as well
https://github.com/adoptium/containers/issues/255

**Comment by pcrosemurgy** [comment]
In my experience `apt` has thrown an error or lengthy warning when trying to run any `apt` command before an `apt update` in a Docker context, so this is odd.

**Comment by Twisol** [comment]
Hooray for mutable infrastructure 😭

**Comment by pcrosemurgy** [comment]
I'm kind of confused about this: https://github.com/NASA-AMMOS/aerie/runs/7797217962?check_suite_focus=true#step:10:916

I updated our `Dockerfile` to run `apt update` but in those logs it doesn't look like the latest `Dockerfile` is being used? Maybe some caching going on, or maybe I'm editing the entirely wrong `Dockerfile`?

**Comment by pcrosemurgy** [comment]
Huh, maybe that test run was from a previous failure.

**Review by skovati — APPROVED** [review]

**Review by patkenneally — APPROVED** [review]

**Pull request merged**